diff --git a/src/c_wrapper/clinfo_ext.h b/src/c_wrapper/clinfo_ext.h
index 6094c52..9263981 100644
--- a/src/c_wrapper/clinfo_ext.h
+++ b/src/c_wrapper/clinfo_ext.h
@@ -4,7 +4,7 @@
 #ifndef _EXT_H
 #define _EXT_H
 
-#ifdef __APPLE__
+#if (defined(__APPLE__) && !defined(PYOPENCL_APPLE_USE_CL_H))
 #include <OpenCL/opencl.h>
 #else
 #include <CL/cl.h>
diff --git a/src/c_wrapper/context.cpp b/src/c_wrapper/context.cpp
index 0a453d0..f2478fd 100644
--- a/src/c_wrapper/context.cpp
+++ b/src/c_wrapper/context.cpp
@@ -68,7 +68,7 @@ context::get_info(cl_uint param_name) const
                 break;
 
 #if defined(PYOPENCL_GL_SHARING_VERSION) && (PYOPENCL_GL_SHARING_VERSION >= 1)
-#if defined(__APPLE__) && defined(HAVE_GL)
+#if defined(__APPLE__) && defined(HAVE_GL) && !defined(PYOPENCL_APPLE_USE_CL_H)
             case CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE:
 #else
             case CL_GL_CONTEXT_KHR:
diff --git a/src/c_wrapper/gl_obj.cpp b/src/c_wrapper/gl_obj.cpp
index 6dfbec8..bd7edf3 100644
--- a/src/c_wrapper/gl_obj.cpp
+++ b/src/c_wrapper/gl_obj.cpp
@@ -137,7 +137,7 @@ have_gl()
 cl_context_properties
 get_apple_cgl_share_group()
 {
-#ifdef __APPLE__
+#if (defined(__APPLE__) && !defined(PYOPENCL_APPLE_USE_CL_H))
     #ifdef HAVE_GL
         CGLContextObj kCGLContext = CGLGetCurrentContext();
         CGLShareGroupObj kCGLShareGroup = CGLGetShareGroup(kCGLContext);
diff --git a/src/c_wrapper/pyopencl_ext.h b/src/c_wrapper/pyopencl_ext.h
index 4b5e787..cd5d711 100644
--- a/src/c_wrapper/pyopencl_ext.h
+++ b/src/c_wrapper/pyopencl_ext.h
@@ -7,7 +7,7 @@
 
 #else
 
-#ifdef __APPLE__
+#if (defined(__APPLE__) && !defined(PYOPENCL_APPLE_USE_CL_H))
 
 #include <OpenCL/opencl.h>
 
diff --git a/src/c_wrapper/wrap_cl.h b/src/c_wrapper/wrap_cl.h
index b097d12..21ff9c0 100644
--- a/src/c_wrapper/wrap_cl.h
+++ b/src/c_wrapper/wrap_cl.h
@@ -13,7 +13,7 @@
 
 #define CL_USE_DEPRECATED_OPENCL_1_1_APIS
 
-#ifdef __APPLE__
+#if (defined(__APPLE__) && !defined(PYOPENCL_APPLE_USE_CL_H))
 
 // {{{ Mac
 
diff --git a/src/c_wrapper/wrap_constants.cpp b/src/c_wrapper/wrap_constants.cpp
index 701638b..16b1d1f 100644
--- a/src/c_wrapper/wrap_constants.cpp
+++ b/src/c_wrapper/wrap_constants.cpp
@@ -421,7 +421,7 @@ void populate_constants(void(*add)(const char*, const char*, int64_t value))
     ADD_ATTR("context_properties",  ,WGL_HDC_KHR);
     ADD_ATTR("context_properties",  ,CGL_SHAREGROUP_KHR);
 #endif
-#if defined(__APPLE__) && defined(HAVE_GL)
+#if defined(__APPLE__) && defined(HAVE_GL) && !defined(PYOPENCL_APPLE_USE_CL_H)
     ADD_ATTR("context_properties",  ,CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE);
 #endif /* __APPLE__ */
 #ifdef CL_CONTEXT_OFFLINE_DEVICES_AMD
