diff --git a/pyopencl/__init__.py b/pyopencl/__init__.py
index 5010f1c7..ecb5c938 100644
--- a/pyopencl/__init__.py
+++ b/pyopencl/__init__.py
@@ -866,7 +866,13 @@ class KernelWithCustomEnqueue(type(self)):
             __call__ = enqueue
             set_args = my_set_args
 
-        self.__class__ = KernelWithCustomEnqueue
+        try:
+            self.__class__ = KernelWithCustomEnqueue
+        except TypeError:
+            # __class__ assignment may not work in all cases, due to differing
+            # object layouts. Fall back to bouncing through kernel_call below.
+            self._enqueue = enqueue
+            self.set_args = my_set_args
 
     def kernel_get_work_group_info(self, param, device):
         cache_key = (param, device.int_ptr)
